cmake/modules/FindHOST_TOOLS.cmake

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2022, Nordic Semiconductor ASA
+
+# FindHOST_TOOLS module for locating a set of tools to use on the host for
+# Zephyr development.
+#
+# This module will lookup following tools for Zephyr development:
+# +---------------------------------------------------------------+
+# | Tool               | Required |  Notes:                       |
+# +---------------------------------------------------------------+
+# | Generic C-compiler | Yes      |  Pre-processing of devicetree |
+# | Zephyr-sdk         |          |                               |
+# | gperf              |          |                               |
+# | openocd            |          |                               |
+# | bossac             |          |                               |
+# | imgtool            |          |                               |
+# +---------------------------------------------------------------+
+#
+# The module defines the following variables:
+#
+# 'CMAKE_C_COMPILER'
+# Path to C compiler.
+# Set to 'CMAKE_C_COMPILER-NOTFOUND' if no C compiler was found.
+#
+# 'GPERF'
+# Path to gperf.
+# Set to 'GPERF-NOTFOUND' if gperf was not found.
+#
+# 'OPENOCD'
+# Path to openocd.
+# Set to 'OPENOCD-NOTFOUND' if openocd was not found.
+#
+# 'BOSSAC'
+# Path to bossac.
+# Set to 'BOSSAC-NOTFOUND' if bossac was not found.
+#
+# 'IMGTOOL'
+# Path to imgtool.
+# Set to 'IMGTOOL-NOTFOUND' if imgtool was not found.
+#
+# 'HOST_TOOLS_FOUND'
+#  True if all required host tools was found.
+
+include(extensions)
+
+if(HOST_TOOLS_FOUND)
+  return()
+endif()
+
+# gperf is an optional dependency
+find_program(GPERF gperf)
+
+# openocd is an optional dependency
+find_program(OPENOCD openocd)
+
+# bossac is an optional dependency
+find_program(BOSSAC bossac)
+
+# imgtool is an optional dependency (the build may also fall back to scripts/imgtool.py
+# in the mcuboot repository if that's present in some cases)
+find_program(IMGTOOL imgtool)
+
+# Pick host system's toolchain if we are targeting posix
+if("${ARCH}" STREQUAL "posix")
+  if(NOT "${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "llvm")
+    set(ZEPHYR_TOOLCHAIN_VARIANT "host")
+  endif()
+endif()
+
+# Keep XCC_USE_CLANG behaviour for a while.
+if ("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc"
+    AND "$ENV{XCC_USE_CLANG}" STREQUAL "1")
+  set(ZEPHYR_TOOLCHAIN_VARIANT xcc-clang)
+  message(STATUS "XCC_USE_CLANG is deprecated. Please set ZEPHYR_TOOLCHAIN_VARIANT to 'xcc-clang'")
+endif()
+
+if(NOT ZEPHYR_TOOLCHAIN_VARIANT AND
+   (CROSS_COMPILE OR (DEFINED ENV{CROSS_COMPILE})))
+    set(ZEPHYR_TOOLCHAIN_VARIANT cross-compile)
+endif()
+
+find_package(Zephyr-sdk 0.15)
```

Comment:
Move to a variable near the top like in the original e.g.:
```
# This is the minimum required Zephyr SDK version.
set(TOOLCHAIN_ZEPHYR_MINIMUM_REQUIRED_VERSION 0.15)
```

Reply:
actually this was done on purpose.

Originally the reason to use a variable was because the minimum version was used in two places, however the disadvantage of the variable is that you must know its name to find it or do a two step search.

Whereas, with `find_package(Zephyr-sdk <version>)` you can search directly for `Zephyr-sdk` which is the package name, and then update the required version directly there.
Also much more readable when reading code, cause you can read the version immediately.

As bonus info the intention is to deprecate and remove those lines, so toolchain configuration gets aligned:
https://github.com/zephyrproject-rtos/zephyr/blob/afe878b9c7868757e1418ac7584b559bcbda3615/cmake/modules/FindHOST_TOOLS.cmake#L71-L81

and to see if this can be cleaned up as well:
https://github.com/zephyrproject-rtos/zephyr/blob/afe878b9c7868757e1418ac7584b559bcbda3615/cmake/modules/FindHOST_TOOLS.cmake#L64-L69

That will move the `find_package(Zephyr-sdk 0.15) further up, and everything becomes even more readable.
But it takes two Zephyr releases before deprecated code can be removed.